plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/application/importoperation/GeneralProjectImportOperation.java

```diff
@@ -42,8 +41,9 @@ public GeneralProjectImportOperation(File projectDirectory) {
 
 		IWorkspace workspace = ResourcesPlugin.getWorkspace();
 		IProject project = workspace.getRoot().getProject(getProjectDirectory().getName());
-		overwriteExistingProject(project, workspace, monitor);
-		importToWorkspace(getProjectDirectory(), workspace, monitor);
+		if (overwriteExistingProject(project, workspace, monitor)) {
```

Comment:
Do we need to ask user at this moment? Method importToWorkspace() called right after only creates description and causes refresh.  Overwriting does not affect files on disk, so that if user says no, all the same as soon as he refreshes the project, the effect to it will be the same as delete/create done here.
Maybe, if .project, .classpath etc files are not shared and are not replaced, then it may make sense to recreate them (I would then call it other than 'overwrite'), but in this case, if user cancels, we may need to call the refresh of the project.